libs/payments/cart/src/lib/cart.manager.ts

```diff
@@ -64,99 +61,104 @@ export class CartManager {
+        ...cart,
+        id: cart.id.toString('hex'),
+        uid: cart.uid ? cart.uid.toString('hex') : undefined,
+      };
     } catch (error) {
       const cause = error instanceof NotFoundError ? undefined : error;
       throw new CartNotFoundError(id, cause);
     }
   }
 
-  public async updateFreshCart(cart: Cart, items: UpdateCart) {
+  public async updateFreshCart(cart: ResultCart, items: UpdateCart) {
     this.checkActionForValidCartState(cart, 'updateFreshCart');
 
-    cart.setCart({
-      ...items,
-    });
-
     try {
-      return await this.handleUpdates(cart);
+      await updateCart(this.db, cart, {
+        ...items,
+        taxAddress: items.taxAddress
+          ? JSON.stringify(items.taxAddress)
+          : undefined,
+        uid: items.uid ? Buffer.from(items.uid, 'hex') : undefined,
+      });
     } catch (error) {
       const cause = error instanceof CartNotUpdatedError ? undefined : error;
       throw new CartNotUpdatedError(cart.id, items, cause);
     }
   }
 
-  public async finishCart(cart: Cart, items: FinishCart) {
+  public async finishCart(cart: ResultCart, items: FinishCart) {
     this.checkActionForValidCartState(cart, 'finishCart');
 
-    cart.setCart({
-      state: CartState.SUCCESS,
-      ...items,
-    });
-
     try {
-      return await this.handleUpdates(cart);
+      await updateCart(this.db, cart, {
+        ...items,
+        uid: items.uid ? Buffer.from(items.uid, 'hex') : undefined,
+        state: CartState.SUCCESS,
+      });
     } catch (error) {
       const cause = error instanceof CartNotUpdatedError ? undefined : error;
       throw new CartNotUpdatedError(cart.id, items, cause);
     }
   }
 
-  public async finishErrorCart(cart: Cart, items: FinishErrorCart) {
+  public async finishErrorCart(cart: ResultCart, items: FinishErrorCart) {
     this.checkActionForValidCartState(cart, 'finishErrorCart');
 
-    cart.setCart({
-      state: CartState.FAIL,
-      ...items,
-    });
-
     try {
-      return await this.handleUpdates(cart);
+      await updateCart(this.db, cart, {
+        ...items,
+        uid: items.uid ? Buffer.from(items.uid, 'hex') : undefined,
+        state: CartState.FAIL,
+      });
     } catch (error) {
       const cause = error instanceof CartNotUpdatedError ? undefined : error;
       throw new CartNotUpdatedError(cart.id, items, cause);
     }
   }
 
-  public async deleteCart(cart: Cart) {
+  public async deleteCart(cart: ResultCart) {
     this.checkActionForValidCartState(cart, 'deleteCart');
 
     try {
-      const result = await cart.delete();
-      if (!result) {
+      if (!(await deleteCart(this.db, cart))) {
         throw new CartNotDeletedError(cart.id);
-      } else {
-        return result;
       }
+      return true;
     } catch (error) {
       const cause = error instanceof CartNotDeletedError ? undefined : error;
       throw new CartNotDeletedError(cart.id, cause);
     }
   }
-
-  public async restartCart(cart: Cart) {
```

Comment:
Just a question. I noticed this method got removed. Did it make more sense for the CartService?

Reply:
Yep, I think deleting or making a new one can be wrapped up there.